### PR TITLE
Display oci helm repositories well

### LINKED ISSUE #2266: SPIKE: Behavior of OCI Helm Chart in Weave GitOps
As a user,
I would like Helm with OCI support in Weave GitOps
so that I could see which Helm Release was installed from which OCI image.

## Technical bits

The new version of HelmRepository will support ".spec.type: oci" to indicate that the source of a HelmRelease will be an OCI image.

image

### PR DESCRIPTION
**Pull request by ozamosi**
This upgrades the source controller, exports the helm repository type, and then displays it.

It also disables linking to oci URLs, and handles helm repositories that don't have a last updated time in both the source table and the details page - there's more details why in the UI commit.

End result:

Screenshot from 2022-06-14 18-33-06

Screenshot from 2022-06-14 18-33-20

This closes #2266

### CODE REVIEW

**Review by joshri — COMMENTED** [review]
Do you know why the Breadcrumbs and Metadata snapshots changed? Seems like you didn't change anything that would effect those

**Comment by opudrovs** [comment]
@joshri good catch! Fixed the tests and snapshots.

File: ui/components/HelmRepositoryDetail.tsx [hunk 1]
```diff
@@ -45,6 +44,18 @@ function HelmRepositoryDetail({
   );
 }
 
+function tryLink(url: string) {
```
Comment (opudrovs) [inline ui/components/HelmRepositoryDetail.tsx:47]: Just a suggestion, maybe we could start adding utility/helper function return types (if not tests). Smth. like `React.ReactElement<any, any> | string` could work.

**Review by opudrovs — APPROVED** [review]
The UI part looks good to me, tested it locally. Also looked through the backend part, looks good, but I don't know why the tests are failing.

File: ui/components/SourcesTable.tsx [hunk 2]
```diff
@@ -108,8 +108,10 @@ function SourcesTable({ className, sources }: Props) {
                 return "-";
               case FluxObjectKind.KindHelmRepository:
                 text = (s as HelmRepository).url;
-                url = text;
-                link = true;
+                if ((s as HelmRepository).repositoryType != "OCI") {
```
Comment (chanwit) [inline ui/components/SourcesTable.tsx:111]: should this be a constant, please?
maybe `HelmRepositoryTypeOCI` ?

**Review by chanwit — APPROVED** [review]

**Pull request merged**